internal/controller/gateway_config_test.go

```diff
@@ -0,0 +1,404 @@
+			Namespace: "default",
+			Annotations: map[string]string{
+				GatewayConfigAnnotationKey: "test-config",
+			},
+		},
+		Spec: gwapiv1.GatewaySpec{
+			GatewayClassName: "test-class",
+			Listeners: []gwapiv1.Listener{
+				{
+					Name:     "http",
+					Port:     8080,
+					Protocol: gwapiv1.HTTPProtocolType,
+				},
+			},
+		},
+	}
+	err = fakeClient.Create(t.Context(), gateway)
+	require.NoError(t, err)
+
+	// Reconcile should not block deletion and should notify the referencing Gateway.
+	_, err = c.Reconcile(t.Context(), reconcile.Request{
+		NamespacedName: client.ObjectKey{Name: "test-config", Namespace: "default"},
+	})
+	require.NoError(t, err)
+
+	events := eventCh.RequireItemsEventually(t, 1)
+	require.Len(t, events, 1)
+}
+
+func TestGatewayConfigController_ReconcileNotFound(t *testing.T) {
+	fakeClient := requireNewFakeClientForGatewayConfig(t)
+	eventCh := internaltesting.NewControllerEventChan[*gwapiv1.Gateway]()
+	c := NewGatewayConfigController(fakeClient, ctrl.Log, eventCh.Ch)
+
+	result, err := c.Reconcile(t.Context(), reconcile.Request{
+		NamespacedName: client.ObjectKey{Name: "missing-config", Namespace: "default"},
+	})
+	require.NoError(t, err)
+	require.Equal(t, ctrl.Result{}, result)
+	require.Empty(t, eventCh.RequireItemsEventually(t, 0))
+}
+
+func TestGatewayConfigController_ListErrorSetsNotAcceptedStatus(t *testing.T) {
+	fakeClient := requireNewFakeClientForGatewayConfig(t)
+	eventCh := internaltesting.NewControllerEventChan[*gwapiv1.Gateway]()
+	errClient := &errorListClient{
+		Client:  fakeClient,
+		listErr: errors.New("list failure"),
+	}
+	c := NewGatewayConfigController(errClient, ctrl.Log, eventCh.Ch)
+
+	gatewayConfig := &aigv1a1.GatewayConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "list-error-config",
+			Namespace: "default",
+		},
+		Spec: aigv1a1.GatewayConfigSpec{},
+	}
+	err := fakeClient.Create(t.Context(), gatewayConfig)
+	require.NoError(t, err)
+
+	_, err = c.Reconcile(t.Context(), reconcile.Request{
+		NamespacedName: client.ObjectKey{Name: "list-error-config", Namespace: "default"},
+	})
+	require.Error(t, err)
+
+	var updated aigv1a1.GatewayConfig
+	err = fakeClient.Get(t.Context(), client.ObjectKey{Name: "list-error-config", Namespace: "default"}, &updated)
+	require.NoError(t, err)
+	require.Len(t, updated.Status.Conditions, 1)
+	require.Equal(t, aigv1a1.ConditionTypeNotAccepted, updated.Status.Conditions[0].Type)
+	require.Equal(t, metav1.ConditionFalse, updated.Status.Conditions[0].Status)
+	require.Contains(t, updated.Status.Conditions[0].Message, "failed to find referencing Gateways")
+}
+
+func TestGatewayConfigController_GatewayReferencesNonExistingConfig(t *testing.T) {
+	fakeClient := requireNewFakeClientForGatewayConfig(t)
+	eventCh := internaltesting.NewControllerEventChan[*gwapiv1.Gateway]()
+	c := NewGatewayConfigController(fakeClient, ctrl.Log, eventCh.Ch)
+
+	// Create a Gateway that references a GatewayConfig that doesn't exist yet.
```

Comment:
This is not what I meant. The thing we're missing in the tests is the behaviour when the referenced config does not exist (and will never exist). For example, users make a typo in the annotation value. What is the expected behaviour then? This is what I think this test needs to validate.